pkg/agent/controller/agent.go

```diff
@@ -219,6 +269,55 @@ func (a *Controller) serviceToRemoteServiceImport(obj runtime.Object, op syncer.
 	return serviceImport, false
 }
 
+func (a *Controller) endpointToRemoteServiceImport(obj runtime.Object, op syncer.Operation) (runtime.Object, bool) {
+	ep := obj.(*corev1.Endpoints)
+	svcExport, err := a.lighthouseClient.LighthouseV2alpha1().ServiceExports(ep.Namespace).Get(ep.Name, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		// Service Export not created yet
+		return nil, false
+	} else if err != nil {
+		// some other error. Log and requeue
+		klog.Errorf("Unable to get ServiceExport for %#v: %v", ep, err)
+		return nil, true
+	}
+
+	serviceImport, err := a.getServiceImport(svcExport)
+
+	if err != nil {
+		// Log and requeue
+		klog.Errorf("Failed to get ServiceImport for %v", svcExport)
+		return nil, true
+	}
+
+	if serviceImport.Spec.Type == lighthousev2a1.Headless && a.globalnetEnabled {
+		klog.Errorf("Headless Services not supported with globalnet yet")
+		return nil, false
+	}
+
+	oldStatus := serviceImport.Status.DeepCopy()
+	ipList := getIpsFromEndpoint(ep)
+
+	if serviceImport.Spec.Type == lighthousev2a1.SuperclusterIP && len(ipList) > 0 {
+		return nil, false
+	}
```

Comment:
I'm not clear on this check. Shouldn't this code only run for `Headless`? For `SuperclusterIP` we use the service's `ClusterIP` (or GN IP) for the `ServiceImport` IP(s). With this check, we would overwrite the IPs with an empty list.